src/feasibility_form_nls.jl

```diff
@@ -339,10 +339,15 @@ function jtprod_residual!(nlp :: FeasibilityFormNLS, x :: AbstractVector, v :: A
   return Jtv
 end
 
+# the Hessian of the residual is
+# [ 0  0 ]  } n
+# [ 0  I ]  } m
 function hess_residual(nlp :: FeasibilityFormNLS, x :: AbstractVector, v :: AbstractVector)
   increment!(nlp, :neval_hess_residual)
   n = nlp.meta.nvar
-  return spzeros(n, n)
+  m = nlp.internal.nls_meta.nequ
+  rows = collect((n-m+1):n)
+  return sparse(rows, rows, ones(eltype(x), m), n, n)
```

Comment:
The residual function, in this case, is `F(x,r) = r`, so the Hessian should really be 0.

Reply:
Maybe I'm confused as to what `hess_residual` is supposed to return. Is it the sum of the Hessians of the individual residuals?

Reply:
Yes.